src/Desc.cc

```diff
@@ -249,6 +257,28 @@ size_t ODesc::StartsWithEscapeSequence(const char* start, const char* end)
 	return 0;
 	}
 
+size_t check_utf8 (const char* bytes, size_t n, size_t i)
```

Comment:
This method can be changed to just use `getNumBytesForUTF8` and not have to loop continually calling `isLegalUTF8Sequence`. The first byte tells you how long the next UTF8 sequence should be.